public/request/message/read.php

```diff
@@ -1,19 +1,21 @@
 <?php
 
-require_once __DIR__ . '/../../../vendor/autoload.php';
-require_once __DIR__ . '/../../../lib/bootstrap.php';
+use Illuminate\Support\Facades\Validator;
 
-if (!ValidatePOSTChars("um")) {
-    exit;
+if (!authenticateFromCookie($user, $permissions, $userDetails)) {
+    abort(401);
 }
 
-$user = requestInputPost('u');
-$messageID = requestInputPost('m', null, 'integer');
+$input = Validator::validate(request()->post(), [
+    'message' => 'required|integer',
+    'status' => 'required|integer|in:0,1',
+]);
 
-$messageReadStatus = requestInputPost('r', 0);    // normally set as read
-
-if (markMessageAsRead($user, $messageID, $messageReadStatus)) {
-    echo "OK:" . $messageID;
-} else {
-    echo "FAILED!";
+if (markMessageAsRead($user, $input['message'], $input['status'])) {
+    if ((int) $input['status'] === 1) {
+        return response()->json(['message' => __('legacy.success.ok')]);
+    }
+    return true;
```

Comment:
Not sure if it's related (or even if it's new behavior), but opening an unread message does not remove the underline indicating that it's now been read. Refreshing the page does reflect the change.

Reply:
It's related - has to do with how the global XHR success and error handler work: https://github.com/RetroAchievements/RAWeb/blob/v2/public/js/all.js#L16-L38

To unify how "ajax" calls react to json responses I added those global handlers so info, success and error messages are displayed at any time that they are provided in a response.

The async "read" action would cause a success status message each time a message detail is opened. To prevent this I chose to return true which results in an empty yet successful HTTP response without triggering the status message. If that's desired it's easy to change. Error responses would still show every time.

The code that changed the looks of read messages on the fly was very convoluted. I did not want to spend too much time with refactoring it so I felt that seeing the correct status after a page reload is enough. I can see how that's a step back in visual feedback quality. I'll see if I can find a simple solution for that at another time; noted.

Reply:
Cleaned up and visual feedback for read and unread is back: https://github.com/RetroAchievements/RAWeb/pull/1103/commits/17483067b981bf7f116a6e3b250534ef8277f2bc
Fixed another issue which prevented deletion of messages where the username case did not match (in my case "Luchaos" vs "luchaos")